frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/disk-inventory.spec.ts

```diff
@@ -0,0 +1,121 @@
+const fetchLocalVolumeDiscovery = () =>
+  cy.exec(`kubectl get --ignore-not-found localvolumediscoveryresults -n openshift-storage -A -o json`);
+
+describe('Disk list is accessible from Nodes view', () => {
+  const diskStatusCount = {
+    available: 0,
+    notAvailable: 0,
+    unknown: 0,
+  };
+
+  const localvolumediscoveryresults = {
+    name: null,
+    nodeName: null,
+    discoveredDevices: null,
+  };
+
+  const lvdCount = (lvd) => {
+    if (!lvd) return diskStatusCount;
+    for (var i = 0; i < lvd.length; i++) {
+      switch(lvd[i].status.state) {
+        case "Available":
+          diskStatusCount.available += 1;
+          break;
+        case "NotAvailable":
+          diskStatusCount.notAvailable += 1;
+          break;
+        case "Unknown":
+          diskStatusCount.unknown += 1;
+      }
+    }
+    return diskStatusCount;
+  }
+
+  const toggleDiskSelect = (diskStatus: string) => {
+    cy.byLegacyTestID('filter-dropdown-toggle').click();
+    cy.byTestRowFilter(diskStatus).parent().click();
+  };
+
+  const testDiskSelection = (diskStatus: string) => {
+    var count = 0;
+    toggleDiskSelect(diskStatus);
+    switch(diskStatus) {
+      case "Available":
+        count = diskStatusCount.available;
+        break;
+      case "NotAvailable":
+        count = diskStatusCount.notAvailable;
+        break;
+      case "Unknown":
+        count = diskStatusCount.unknown;
+    }
+    if (count == 0) {
+      cy.byTestRows("resource-row").should('not.exist')
+    } else {
+      cy.byTestRows("resource-row").parent().find('tr')
+        .should('have.length', count)
+    }
+    toggleDiskSelect(diskStatus);
```

Comment:
You can avoid switch case here.
Since you only need to check the resource -rows count to be equal to the disk state count .
So you can just pass the disk state count directly in the function params and check for that particular state.

Reply:
Sure will do as follows:
 const testDiskSelection = (diskStatus: string, diskStatusCount: number) => {
  toggleDiskSelect(diskStatus);
  if (diskStatusCount == 0) {
    cy.byTestRows("resource-row").should('not.exist')
  } else {
    cy.byTestRows("resource-row").parent().find('tr')
      .should('have.length', diskStatusCount)
  }
  toggleDiskSelect(diskStatus);
}